simulation/src/xline_group.rs

```diff
@@ -0,0 +1,129 @@
+use std::{collections::HashMap, sync::Arc, time::Duration};
+
+use curp::members::{ClusterInfo, ServerId};
+use itertools::Itertools;
+use madsim::runtime::NodeHandle;
+use tracing::debug;
+use utils::config::{ClientConfig, CompactConfig, CurpConfig, ServerTimeout, StorageConfig};
+use xline::{server::XlineServer, storage::db::DB};
+use xline_client::{error::XlineClientError, types::kv::PutRequest, Client, ClientOptions};
+use xlineapi::{command::Command, PutResponse};
+
+struct XlineNode {
+    addr: String,
+    handle: NodeHandle,
+}
+
+pub struct XlineGroup {
+    nodes: HashMap<ServerId, XlineNode>,
+    client_handle: NodeHandle,
+}
+
+impl XlineGroup {
+    pub async fn new(size: usize) -> Self {
+        assert!(size >= 3, "the number of nodes must >= 3");
+        let handle = madsim::runtime::Handle::current();
+
+        let all: HashMap<_, _> = (0..size)
+            .map(|x| (format!("S{x}"), vec![format!("192.168.1.{}:12345", x + 1)]))
+            .collect();
+        let nodes = (0..size)
+            .map(|i| {
+                let name = format!("S{i}");
+                let addr = format!("192.168.1.{}:12345", i + 1);
+                let cluster_info = Arc::new(ClusterInfo::new(all.clone(), &name));
+                let id = cluster_info.self_id();
+
+                let handle = handle
+                    .create_node()
+                    .name(id.to_string())
+                    .ip(format!("192.168.1.{}", i + 1).parse().unwrap())
+                    .init(move || {
+                        let server = XlineServer::new(
+                            cluster_info.clone(),
+                            i == 0,
```

Comment:
we might restart the old leader.
after the reboot, it may no longer be the leader.